clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2766,11 +2772,13 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
                                        const CXXRecordDecl *KernelObj) {
     TypeSourceInfo *TSInfo =
         KernelObj->isLambda() ? KernelObj->getLambdaTypeInfo() : nullptr;
-    VarDecl *VD = VarDecl::Create(
-        Ctx, DC, KernelObj->getLocation(), KernelObj->getLocation(),
-        KernelObj->getIdentifier(), QualType(KernelObj->getTypeForDecl(), 0),
-        TSInfo, SC_None);
+    IdentifierInfo *Ident = KernelObj->getIdentifier();
+    if (!Ident)
+      Ident = &Ctx.Idents.getOwn("__SYCLKernel");
```

Comment:
I still don't believe you need to do 'getOwn', I think 'get' is more than sufficient.    getOwn I believe is for cases where the identifier comes form an external file.

otherwise I think this approach is fine.  I'll leave it up to the others to review the rest of this.

Reply:
Done. Thank you for insightful comments!